extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -157,15 +157,20 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
    * @private
    */
   onShoppingDataUpdate_(shoppingData) {
-    if (!shoppingData.activeProductData || !this.isOnActivePage_()) {
-      return;
-    }
     const isOpen = this.storeService_.get(StateProperty.PAGE_ATTACHMENT_STATE);
+
     if (isOpen) {
       // If open, update the template.
-      // This happens when a product card is clicked in the PLP template.
+      // This happens when a product card is clicked in the PLP template
+      // or when the shopping attachment is opened when product data is loaded.
       this.updateTemplate_(shoppingData);
-    } else {
+    }
+
+    if (!shoppingData.activeProductData || !this.isOnActivePage_()) {
+      return;
+    }
```

Comment:
Without checking if we're on the active page first, code will run on all pages with this component.